docs/docs/adding-search-with-js-search.md

```diff
@@ -1,8 +1,357 @@
+            },
+          },
+        })
+        resolve()
+      })
+      .catch(err => {
+        console.log("====================================")
+        console.log(`error creating Page:${err}`)
+        console.log("====================================")
+        reject(new Error(`error on page creation:\n${err}`))
+      })
+  })
+}
+```
+
+Contrary to our earlier approach, instead of letting the component do all of the work, it's Gatsby's job to do the work and pass all the data to a page defined by the path object, via [pageContext](https://www.gatsbyjs.org/docs/behind-the-scenes-terminology/#pagecontext).
+
+Let's do this by adding a couple of new components. The first will be your page template component.
+
+It will be created in the following location `/src/templates`, under the name `ClientSearchTemplate.js`, with the following code:
+
+```javascript
+import React from "react"
+import Layout from "../components/layout"
+import ClientSearch from "../components/ClientSearch"
+
+const SearchTemplate = props => {
+  const { pageContext } = props
+  const { bookData } = pageContext
+  const { allBooks, options } = bookData
+  return (
+    <Layout>
+      <h1 style={{ marginTop: "3em", textAlign: "center" }}>
+        Search data using JS Search using Gatsby Api
+      </h1>
+      <h3 style={{ marginTop: "2em", padding: "2em 0em", textAlign: "center" }}>
+        Books Indexed by:
+      </h3>
+
+      <div>
+        <ClientSearch books={allBooks} engine={options} />
+      </div>
+    </Layout>
+  )
+}
+
+export default SearchTemplate
+```
+
+Now add the second component. Add a file called `ClientSearch.js` to the `components` folder, with the following code as a baseline:
+
+```javascript
+import React, { Component } from 'react'
+import { Segment, Form, Header } from 'semantic-ui-react'
+import * as JsSearch from 'js-search'
+import DataTable from './DataTable'
+import './search.css'
+class ClientSearch extends Component {
+  state = {
+    isLoading: true,
+    searchResults: [],
+    search: null,
+    isError: false,
+    indexByTitle: false,
+    indexByAuthor: false,
+    termFrequency: true,
+    removeStopWords: false,
+    searchQuery: '',
+    selectedStrategy: '',
+    selectedSanitizer: '',
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (prevState.search === null) {
+      const { engine } = nextProps
+      return {
+        indexByTitle: engine.TitleIndex,
+        indexByAuthor: engine.AuthorIndex,
+        termFrequency: engine.SearchByTerm,
+        selectedSanitizer: engine.searchSanitizer,
+        selectedStrategy: engine.indexStrategy,
+      }
+    }
+    return null
+  }
+  async componentDidMount() {
+    this.rebuildIndex()
+  }
+  .....
+}
+export default ClientSearch
+```
+
+The code used in here is almost identical to the one documented in the first approach and for the same reason stated in the first approach, here also the full implementation will not be shown.
```

Comment:
Similar to my earlier comment, let's make it very clear how to get from this step of the tutorial to a working local site:

- combine files if possible
- clearly describe which files should be copied to your local site
- mention starting the site with `gatsby develop` and the url to visit